### ISSUE TITLE
[Feature Request]: Add Typescript typings

### ISSUE THREAD

**Issue by HaaLeo** [M-Scott-Lassiter/Alphanumeric-Encoder#43]
### Contact Details

@GitHub

### Description

I would like to use this NPM package in a typescript project. To enable intellisense for IDEs the package should ship with its typescript typings. How this can be done is documented [here](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html)

### Are you able/willing to make the change? (It's ok if you're not!)

No

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/M-Scott-Lassiter/Alphanumeric-Encoder/blob/main/CODE_OF_CONDUCT.md)

**Comment by M-Scott-Lassiter** [comment]
@HaaLeo Thanks for your interest in the library! I do not have experience using Typescript, but I gave it a try. I just published a new release under a [beta tag on NPM](https://www.npmjs.com/package/alphanumeric-encoder/v/1.6.0-beta.1).

Would you mind taking a look at it and see if it successfully gave you the bindings you were looking for?

**Comment by HaaLeo** [comment]
The package seems to not include any typings 🤔 :
![image](https://user-images.githubusercontent.com/23164374/169615661-ac07ee20-7a15-4463-8dfb-1950b6d0e69f.png)

**Comment by M-Scott-Lassiter** [comment]
> The package seems to not include any typings 🤔 

Yeah, that's my fault. I neglected to add the index.d.ts file to the "files" array in package.json. I think I've fixed it in beta.2 now. Does that work?

```
npm i alphanumeric-encoder@1.6.0-beta.2
```

**Comment by M-Scott-Lassiter** [comment]
@HaaLeo  Gentle reminder, 1.6.0-beta.2 is available. Have you had a chance to review it to check if it gives you the Typescript functionality?

**Comment by HaaLeo** [comment]
@M-Scott-Lassiter thx for the reminder. I should be able to review it on Monday. I'll keep you updated.

**Comment by HaaLeo** [comment]
I tested `1.6.0-beta.2`: When I import the package like 
```
import { AlphanumericEncoder } from 'alphanumeric-encoder';
```

I get all the intelisene and so on. However, at runtime I get the following error:
```
alphanumeric_encoder__WEBPACK_IMPORTED_MODULE_7__.AlphanumericEncoder is not a constructor
```

But when I import the package like this:
```
import AlphanumericEncoder from 'alphanumeric-encoder';
```

it works as expected at runtime. However the intellisense shows an error:
```
This expression is not constructable.
  Type 'typeof import("/Users/z0049e1t/_GIT/Siemens/bsp1/pmd-backend/node_modules/alphanumeric-encoder/index")' has no construct signatures.ts(2351)
```

**Comment by M-Scott-Lassiter** [comment]
@HaaLeo Based on some feedback I got on that StackOverflow question, I made some more changes. Although I am not using Typescript, I did turn on VS Code's `// @ts-check` in the index.js file and the index.test.js file. I saw the errors you were talking about when I did that, and after this push I no longer see them. Fingers crossed, I think this might have fixed it.

When you can, would you mind trying this new beta version?

```bash
npm i alphanumeric-encoder@1.6.0-beta.3
```

**Comment by HaaLeo** [comment]
@M-Scott-Lassiter  `1.6.0-beta.3` seems to do the job 🙌🏼.

**Comment by M-Scott-Lassiter** [comment]
@HaaLeo I merged it into the main branch with PR #45. Released as [`v1.6.0`](https://www.npmjs.com/package/alphanumeric-encoder).

```bash
npm i alphanumeric-encoder
```

Thanks for your help and support on this!
